src/Bbt.Maybe/Maybe.cs

```diff
@@ -0,0 +1,243 @@
+        public static bool operator ==(Maybe<T> aA, Maybe<T> aB)
+        {
+            return object.Equals(aA.mValue, aB.mValue);
+        }
+
+        /// <summary>
+        /// Checks whether the operands are unequal.
+        /// </summary>
+        /// <param name="aA">Maybe to compare.</param>
+        /// <param name="aB">Maybe to compare.</param>
+        public static bool operator !=(Maybe<T> aA, Maybe<T> aB)
+        {
+            return !(aA == aB);
+        }
+
+        /// <summary>
+        /// Executes <paramref name="aDoAction"/> if value is not null.
+        /// </summary>
+        /// <param name="aDoAction">The action which is performed if maybe is not none.</param>
+        /// <returns>The none case.</returns>
+        public NoneCase Do(Action<T> aDoAction)
+        {
+            if (aDoAction == null)
+            {
+                throw new ArgumentNullException(nameof(aDoAction));
+            }
+
+            if (this.mValue != null)
+            {
+                aDoAction(this.mValue);
+                return new NoneCase(false);
+            }
+
+            return new NoneCase(true);
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Maybe{TResult}"/> of the projection function <paramref name="aFunc"/>.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the projected result.</typeparam>
+        /// <param name="aFunc">The projection function.</param>
+        /// <returns>The projected maybe.</returns>
+        public Maybe<TResult> Some<TResult>(Func<T, TResult> aFunc)
+            where TResult : class
+        {
+            if (aFunc == null)
+            {
+                throw new ArgumentNullException(nameof(aFunc));
+            }
+
+            var lMaybe = Maybe.None<TResult>();
+            this.Do(aX => lMaybe = Maybe.Some(aFunc(aX)));
+
+            return lMaybe;
+        }
+
+        /// <summary>
+        /// Returns the value in case it is initialized.
+        /// Otherwise throws an <see cref="InvalidOperationException"/>.
+        /// </summary>
+        /// <param name="aMaybeParameterName">The maybe reference used in error message.</param>
+        /// <param name="aAdditionalMessage">Additional error message.</param>
+        /// <returns>The value.</returns>
+        public T ThrowExceptionIfNone(
+            string aMaybeParameterName,
+            string aAdditionalMessage = "")
+        {
+            if (!this.HasValue)
+            {
+                var lMsg = $"({aMaybeParameterName}) may not be null."
+                    .ToString(CultureInfo.InvariantCulture);
+                if (!string.IsNullOrEmpty(aAdditionalMessage))
+                {
+                    lMsg = lMsg + " " + aAdditionalMessage;
+                }
+
+                throw new InvalidOperationException(lMsg);
+            }
+
+            return this.mValue;
+        }
+
+        /// <summary>
+        /// See <see cref="object.Equals(object)"/>.
```

Comment:
We can use `<inheritdoc />` here. Together with a tool which can resolve them it would copy documentation from inherited member during build time. But can also be done in a subsequent issue/PR.

Reply:
Created #12 for tracking this